### PR TITLE
DSA implemations of several Programming languages

### LINKED ISSUE #796: Add Data Structures and Algorithms (DSA) Implementations for Each Programming Language
Enhance the repository by adding [Data Structures and Algorithms (DSA)](https://github.com/jfmartinz/ResourceHub/tree/main/Data%20Structures%20and%20Algorithm%20(DSA)) implementations for various programming languages. This will provide a comprehensive resource for learning and reference.

Create a new section called "DSA by Language," with each table containing different DSA categories like arrays, linked lists, stacks, queues, trees, graphs, etc.

Each implementation should include:

Common data structures (e.g., arrays, linked lists, stacks, queues, trees, graphs)

### PR DESCRIPTION
**Pull request by Pujan-sarkar**
# Pull Request

## Description

This repository contains a collection of Data Structures and Algorithms (DSA) implementations across various programming languages. Each section is organized by language and includes common DSA topics such as arrays, strings, linked lists, trees, and more.  

Fixes #796 

## Type of change
- [X] Adding a new Resource
- [X] Adding a new Category
- [X] Documentation enhancement or fixes
- [X] Website enhancement or fixes

## Checklist:
- [X] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Pujan-sarkar** [comment]
@jfmartinz  please check this pr and merge .

**Comment by Pujan-sarkar** [comment]
@jfmartinz  please merge the pr please

**Comment by jfmartinz** [comment]
You don't need to add new readme, just add it [here](https://github.com/jfmartinz/ResourceHub/tree/main/Data%20Structures%20and%20Algorithm%20(DSA))

Follow this format, each language should have their own table
image

**Comment by Pujan-sarkar** [comment]
@jfmartinz I add a new folder please level 3 or level 2 .

**Comment by jfmartinz** [comment]
Follow this format for adding a new table:

https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md

**Comment by jfmartinz** [comment]
Add it into a table. Each programing language shoukld have a table each

**Comment by Pujan-sarkar** [comment]
@jfmartinz okay tell me i have to add the dsa table in the three programming language's folder ?

**Comment by Pujan-sarkar** [comment]
@jfmartinz I have to add the table of every DSA of every language in the DSA folder ?????

**Comment by Pujan-sarkar** [comment]
@jfmartinz please check I have done the changes as you said earlier.

File: Data Structures and Algorithm (DSA)/readme.md [hunk 1]
```diff
@@ -15,9 +15,14 @@ This section serves as a curated collection of valuable resources for learning a
 - [Youtube Channels](#youtube-channels)
 - [Coding Platforms](#coding-platforms)
 - [Practice Sheets](#sheets)
+- [DSA implementation](#dsa-implementation-of-several-programming-languages)
+   -[C++](#c)
+   -[Java](#java)
+   -[Python](#python)
+   -[Javascript](#javascript)
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:22]:
```suggestion
   -[C++](#c)<br>
   -[Java](#java)<br>
   -[Python](#python)<br>
   -[Javascript](#javascript)<br>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]
Follow this format for table:

### C++
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">String</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>

### JavaSCript
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">String</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>

**Comment by Pujan-sarkar** [comment]
@jfmartinz sir check now , I think this is good , so please merge it.

**Comment by jfmartinz** [comment]
Follow the format. Each table should have only 2 cols, Resource Name and Description.

File: Data Structures and Algorithm (DSA)/readme.md [hunk 2]
```diff
@@ -15,9 +15,16 @@ This section serves as a curated collection of valuable resources for learning a
 - [Youtube Channels](#youtube-channels)
 - [Coding Platforms](#coding-platforms)
 - [Practice Sheets](#sheets)
+- [DSA implementation](#dsa-implementation-of-several-programming-languages)
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:18]:
```suggestion
- [DSA by language](#dsa-by-language)
```

File: Data Structures and Algorithm (DSA)/readme.md [hunk 3]
```diff
@@ -226,7 +233,93 @@ To the Point, Short and Practical - These 3 terms defines our Content
...
+| **Graph**              | [JavaScript Graph Implementation](https://github.com/TheAlgorithms/Javascript/blob/master/Data-Structures/Graphs/graph.js) | Example code for implementing graphs in JavaScript.              |
+|                        | [JavaScript Graph Algorithms](https://www.geeksforgeeks.org/graph-data-structure-and-algorithms/) | Guide on various graph algorithms.                               |
+| **Sorting Algorithms** | [JavaScript Sorting Algorithms](https://github.com/TheAlgorithms/Javascript/tree/master/Sorting)  | Example code for different sorting algorithms in JavaScript.     |
+| **Search Algorithms**  | [JavaScript Search Algorithms](https://github.com/TheAlgorithms/Javascript/tree/master/Searches)  | Example code for different searching algorithms in JavaScript.   |
+| **GitHub Repository**  | [The Algorithms - JavaScript](https://github.com/TheAlgorithms/Javascript)                        | A collection of DSA implementations in JavaScript.               |
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:318]: Follow this format:
### Section name
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">RESOURCE NAME</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>

File: Data Structures and Algorithm (DSA)/readme.md [hunk 4]
```diff
@@ -15,9 +15,16 @@ This section serves as a curated collection of valuable resources for learning a
 - [Youtube Channels](#youtube-channels)
 - [Coding Platforms](#coding-platforms)
 - [Practice Sheets](#sheets)
+- [DSA implementation](#dsa-implementation-of-several-programming-languages)
+
+   -[C++](#c)<br>
+   -[Java](#java)<br>
+   -[Python](#python)<br>
+   -[Javascript](#javascript)<br>
```
Comment (jfmartinz) [inline Data Structures and Algorithm (DSA)/readme.md:23]:
```suggestion
   - [C++](#c)<br>
   - [Java](#java)<br>
   - [Python](#python)<br>
   - [Javascript](#javascript)<br>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by Pujan-sarkar** [comment]
@jfmartinz I guess now mall the changes has done ✌️

**Review by jfmartinz — CHANGES_REQUESTED** [review]
Change it:
image

add space before the '-'. Like this
- C++
-  JS

2. Change the [heading](https://github.com/jfmartinz/ResourceHub/blob/549808fc8d21f3ae9ff74ec7da31c3fd5dc93e0e/Data%20Structures%20and%20Algorithm%20(DSA)/readme.md#dsa-implementation-of-several-programming-languages) to DSA by language

3. Remove the roadmap col in C++, Java, Python, JS
image


4. I already provided the template for addign new table and still not following it.

Check the format of your table, is your table same thing with this format:

```markdown
### Section name
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">RESOURCE NAME</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>
```

5. Remove these lines between the tables:
image

**Comment by Pujan-sarkar** [comment]
@jfmartinz  If this is not working then unassign me i will work on any other issue

**Review by jfmartinz — APPROVED** [review]
LGTML!

**Pull request merged**